src/controllers/auth.controller.js

```diff
@@ -0,0 +1,131 @@
+import { User } from '../models/user.js';
+import { userServices } from '../services/user.service.js';
+import { jwtService } from '../services/jwt.service.js';
+import { ApiError } from '../exceptions/api.error.js';
+import bcrypt from 'bcrypt';
+import { tokenService } from '../services/token.service.js';
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPatterns = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+
+  if (!emailPatterns.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+const register = async (req, res) => {
+  const { email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
```

Comment:
The `ApiError.badRequest` should be used for validation errors, but ensure that the error messages are clear and specific to the field that failed validation.